openedx/features/content_tagging/rules.py

```diff
@@ -35,72 +45,134 @@ def is_taxonomy_admin(user: User, taxonomy: oel_tagging.Taxonomy = None) -> bool
-    Anyone can view an enabled taxonomy,
-    but only taxonomy admins can view a disabled taxonomy.
+    Everyone can potentially view a taxonomy (taxonomy=None). The object permission will be checked on a
+    second pass to determine if the user can change this specific taxonomy.
+    Only taxonomy admins can view a disabled taxonomy. #ToDo: Check if this is correct.
+    """
+    if not taxonomy:
+        return True
+
+    taxonomy = taxonomy.cast()
+
+    return is_taxonomy_admin(user, taxonomy) or (is_global_taxonomy(taxonomy) and taxonomy.enabled)
+
+
+@rules.predicate
+def can_add_taxonomy(user: User) -> bool:
+    """
+    Only taxonomy admins can add taxonomies.
     """
-    if taxonomy:
-        taxonomy = taxonomy.cast()
-    return (taxonomy and taxonomy.enabled) or is_taxonomy_admin(user, taxonomy)
+    return is_taxonomy_admin(user)
 
 
 @rules.predicate
 def can_change_taxonomy(user: User, taxonomy: oel_tagging.Taxonomy = None) -> bool:
     """
+    Everyone can potentially change a taxonomy (taxonomy=None). The object permission will be checked on a
+    second pass to determine if the user can change this specific taxonomy.
     Even taxonomy admins cannot change system taxonomies.
     """
-    if taxonomy:
-        taxonomy = taxonomy.cast()
-    return is_taxonomy_admin(user, taxonomy) and (
-        not taxonomy or (taxonomy and not taxonomy.system_defined)
-    )
+    if not taxonomy:
+        return True
+
+    taxonomy = taxonomy.cast()
+
+    return is_taxonomy_admin(user, taxonomy) and (not taxonomy.system_defined)
+
+
+@rules.predicate
+def can_add_tag(user: User) -> bool:
+    """
+    Only taxonomy admins can add tags.
+    """
+    return is_taxonomy_admin(user)
 
 
 @rules.predicate
 def can_change_taxonomy_tag(user: User, tag: oel_tagging.Tag = None) -> bool:
     """
-    Even taxonomy admins cannot add tags to system taxonomies (their tags are system-defined), or free-text taxonomies
+    Everyone can potentially change a taxonomy tag (tag=None). The object permission will be checked on a
+    second pass to determine if the user can change this specific tag.
+    Even taxonomy admins cannot edit tags from system taxonomies (their tags are system-defined), or free-text taxonomies
     (these don't have predefined tags).
     """
-    taxonomy = tag.taxonomy if tag else None
-    if taxonomy:
-        taxonomy = taxonomy.cast()
+    if not tag:
+        return True
+
+    if tag.taxonomy:
+        taxonomy = tag.taxonomy.cast()
+    else:
+        taxonomy = None
+
     return is_taxonomy_admin(user, taxonomy) and (
-        not tag
-        or not taxonomy
-        or (taxonomy and not taxonomy.allow_free_text and not taxonomy.system_defined)
+        not taxonomy or (not taxonomy.allow_free_text and not taxonomy.system_defined)
     )
 
+@rules.predicate
+def can_add_object_tag(user: User) -> bool:
+    """
+    Only taxonomy admins can add taxonomy tags.
+    """
+    return is_taxonomy_admin(user)
```

Comment:
This is not true.. Normal content authors must to be able to create object tags (i.e tag their content).

Reply:
Reverted these changes